### PR TITLE
Improve robustness of our handling of file path to source URLs

### PR DESCRIPTION
**Pull request by nirvn**
Superseding https://github.com/opengisch/QField/pull/5527

@boardend , I'm hopeful we're getting there.

### CODE REVIEW

**Comment by qfield-fairy** [comment]
🎉 Ta-daaa, freshly created APKs are available for 89674e35871e489a34e0552b524adb87362fef8a: [**arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-arm64.apk)
<details>
<summary>Other architectures</summary>

- [arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-arm-neon.apk)
- [x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-x64.apk)
- [x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-x86.apk)
- [**all access arm64-android**](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-arm64.apk)
- [all access arm-neon-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-arm-neon.apk)
- [all access x64-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-x64.apk)
- [all access x86-android](https://qfieldapks.sos-ch-dk-2.exoscale-cdn.com/ci-builds/qfield_all_access-dev-5528-eda8f38b1f74b56c02ecb53ea92f1d8ec52da765-android-x86.apk)
</details>

File: src/core/utils/urlutils.h [hunk 1]
```diff
@@ -30,11 +30,13 @@ class QFIELD_CORE_EXPORT UrlUtils : public QObject
...
 
-
     /**
-     * Checks whether the provided string is a relative url (has no protocol or starts with `file://`).
+     * Checks whether the provided string is a relative \a url (has no protocol or starts with `file://`).
      */
     static Q_INVOKABLE bool isRelativeOrFileUrl( const QString &url );
+
+    //! Returns a URL from a \a string with logic to handle local paths
+    static Q_INVOKABLE QUrl fromString( const QString &string );
```
Comment (m-kuhn) [inline src/core/utils/urlutils.h:39]: Can we have a unit test (not spix) for this one?
I think it's quite well defined, what it does.

File: test/spix/smoke_test.py [hunk 2]
```diff
@@ -166,6 +167,52 @@ def test_projection(app, screenshot_path, screenshot_check, extra, process_alive
...
+@pytest.mark.project_file("test_image_attachment.qgz")
+def test_projection(app, screenshot_path, screenshot_check, extra, process_alive):
+    """
+    Starts a test app and check for proper reprojection support (including rendering check and message logs).
+    This also tests that QField is able to reach proj's crucial proj.db
+    """
+    assert app.existsAndVisible("mainWindow")
+
+    # Arbitrary wait period to insure project fully loaded and rendered
+    time.sleep(4)
```
Comment (m-kuhn) [inline test/spix/smoke_test.py:179]: Random note: we will want a signal to catch this state at some point

File: test/spix/smoke_test.py [hunk 3]
```diff
@@ -166,6 +167,52 @@ def test_projection(app, screenshot_path, screenshot_check, extra, process_alive
...
+
+    messagesCount = 0
+    for i in range(0, 10):
+        message = app.getStringProperty(
+            "mainWindow/messageLog/messageItem_{}/messageText".format(i), "text"
```
Comment (m-kuhn) [inline test/spix/smoke_test.py:184]:
```suggestion
            `mainWindow/messageLog/messageItem_${i}/messageText`, "text"
```

I was wondering if this is more javascriptonic (js rookie here !)

**Comment by m-kuhn** [comment]
Amazing work on spix side !!
Can you add a few tests on functional side on url side (this will document the expected behavior of the new method much better than a simple "image shows")

File: src/core/utils/urlutils.h (hunk 1 above)
Comment (nirvn) [reply src/core/utils/urlutils.h:39]: @m-kuhn , fair point, I wasn't even thinking about spix testing this function, just the functionality. I'll add something.

**Review by m-kuhn — APPROVED** [review]

**Pull request merged**